pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -372,39 +372,78 @@ def required_predicates(self) -> List[Predicate]:
 
         return preds
 
-    def rebase_pass(self, implicit_swaps: bool = False) -> BasePass:
+    @property
+    def _two_qubit_gate_set(self) -> Set[OpType]:
+        """
+        Assumes that only possibly supported two-qubit gates are
+        ZZPhase, ZZMax and TK2.
+
+        :return: Set of two-qubit OpType in gateset.
+        :rtype: Set[OpType]
         """
-        :param implicit_swaps: If true, allows rebasing of Circuit via TK2 gates
-            to use implicit wire swaps in circuit construction if it reduces
-            the total 2qb qate count.
-        :type implicit_swaps: bool
+        supported_2qb_gates: Set[OpType] = set()
+        if OpType.ZZPhase in self._gate_set:
+            supported_2qb_gates.add(OpType.ZZPhase)
+        if OpType.ZZMax in self._gate_set:
+            supported_2qb_gates.add(OpType.ZZMax)
+        if OpType.TK2 in self._gate_set:
+            supported_2qb_gates.add(OpType.TK2)
+        return supported_2qb_gates
+
+    def rebase_pass(self, **kwargs: QuumKwargTypes) -> BasePass:
+        """
+        Supported kwargs:
+        * `implicit_swaps`: Boolean flag, which if true, returns
+            rebasing pass that allows implicit wire swaps.
+            Default False.
+        * `target_2qb_gate`: pytket OpType, if provided, will
+            return a rebasing pass that only allows given
+            two-qubit gate type.
         :return: Compilation pass for rebasing circuits
         :rtype: BasePass
         """
-        return auto_rebase_pass(self._gate_set, implicit_swaps)
+        target_2qb_optype: Union[None, OpType] = kwargs.get("target_2qb_gate")
+        if target_2qb_optype is None:
+            return auto_rebase_pass(
+                self._gate_set, allow_swaps=bool(kwargs.get("implicit_swap", True))
+            )
+        elif target_2qb_optype in self._two_qubit_gate_set:
+            return auto_rebase_pass(
+                self._gate_set - self._two_qubit_gate_set | {target_2qb_optype},
+                allow_swaps=bool(kwargs.get("implicit_swap", True)),
+            )
+        else:
+            raise QuantinuumAPIError(
+                "Requested target_2qb_gate is not supported by the given Device. "
+                "Please check _two_qubit_gate_set attribute to see which two-qubit "
+                "gates are supported."
+            )
 
-    def default_compilation_pass(
-        self, optimisation_level: int = 2, implicit_swaps: bool = True
+    def default_compilation_pass_options(
+        self, optimisation_level: int = 2, **kwargs: QuumKwargTypes
     ) -> BasePass:
         """
         :param optimisation_level: Allows values of 0,1 or 2, with higher values
```

Comment:
Maybe rather than repeating the docstrings we should just say "see ..." and just list the supported kwargs (otherwise we end up with duplicated docs which get hard to maintain).